google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldMask.java

```diff
@@ -17,29 +17,57 @@
 package com.google.cloud.firestore;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 import java.util.SortedSet;
 import java.util.TreeSet;
+import javax.annotation.Nonnull;
 
-/** A DocumentMask contains the field paths affected by an update. */
-final class DocumentMask {
-  static final DocumentMask EMPTY_MASK = new DocumentMask(new TreeSet<FieldPath>());
+/** A FieldMask can be used to limit the number of fields returned by a `getAll()` call. */
+public final class FieldMask {
+  static final FieldMask EMPTY_MASK = new FieldMask(new TreeSet<FieldPath>());
 
   private final SortedSet<FieldPath> fieldPaths; // Sorted for testing.
```

Comment:
I don't have strong opinion but is it worth using `ImmutableSortedSet` here?

Reply:
I'll leave as is for now, mainly because I don't yet use any Guava classes. The internal object can already not be edited as is, but this is mostly enforced by convention for now.